tests/test_iteration.py

```diff
@@ -0,0 +1,118 @@
+from __future__ import division
+from math import ceil
+from operator import mul
+from functools import reduce
+from itertools import product
+
+import pytest
+from nchelpers.iteration import chunk_corners, chunk_slices, chunks, \
+    opt_chunk_shape
+
+import numpy as np
+
+
+def size(shape):
+    return reduce(mul, shape, 1)
+
+
+def num_chunks(shape, chunk_shape):
+    """Number of chunks of shape ``chunk_shape`` in an array of shape ``shape``."""
+    return reduce(mul, (ceil(s/c) for s, c in zip(shape, chunk_shape)), 1)
+
+
+def union(sets):
```

Comment:
Wouldn't this be the same as the more succinct set comprehension syntax:

```python
{ s for s in sets }
```

Reply:
No, but this would:

```
{ x for s in sets for x in s }
```